### ISSUE TITLE
Trailing spaces in columns are treated as part of the column name

### ISSUE THREAD

**Issue by laurenceisla** [PostgREST/postgrest#2285]
### Environment

* PostgreSQL version: 14.2
* PostgREST version: 9.0.0.20220211

### Description of issue

The `pField` function does not ignore trailing spaces when a column is selected:
```
/table?select=name,join (*) 
```
It does work for leading spaces though: 
```
/table?select=name, join(*)
``` 
To keep consistency it should either allow both or ignore both (I think the latter is the better option).

**Comment by wolfgangwalther** [comment]
I don't think we should treat spaces in any special way in the URI - they are not reserved in our syntax, right? And I think they are not special for URIs in general, either?

So I think the proper way would be to treat both leading and trailing spaces as part of the name.

**Comment by steve-chavez** [comment]
> So I think the proper way would be to treat both leading and trailing spaces as part of the name

Hm, that would be a big breaking change, many users add whitespace to make the syntax more readable.

**Comment by wolfgangwalther** [comment]
Is that documented anywhere, though?

**Comment by steve-chavez** [comment]
Not documented, but there are tests

https://github.com/PostgREST/postgrest/blob/3648986aa8c347be72dafee09df074b0dde67dcb/test/spec/Feature/Query/AndOrParamsSpec.hs#L195-L197

Some old tests rely on this too

https://github.com/PostgREST/postgrest/blob/3648986aa8c347be72dafee09df074b0dde67dcb/test/spec/Feature/Query/QuerySpec.hs#L375-L378
